src/common/git/husky.ts

```diff
@@ -1,23 +1,64 @@
 import * as path from 'path'
 import type {NodeProject} from 'projen/lib/javascript'
 import {AssetFile} from '../files/AssetFile'
-import type {WithGitHooks} from './with-git-hooks'
+import {CheckCargoOptions, HuskyRule, WithGitHooks} from './with-git-hooks'
 
 export const addHusky = (project: NodeProject, options: WithGitHooks): void => {
   project.addDevDeps('husky')
   project.addTask('prepare', {exec: 'husky install'})
+  const huskyRules = options.huskyRules ?? [HuskyRule.COMMIT_MESSAGE]
 
-  if (options.hasDefaultCommitHook === false) {
-    return
+  huskyRules.forEach((ruleId) => {
+    switch (ruleId) {
+      case HuskyRule.CHECK_CARGO:
+        const checkCargoOptions = options.checkCargoOptions ?? {}
+        checkCargo(project, checkCargoOptions)
+        break
+      case HuskyRule.COMMIT_MESSAGE:
+        commitMessage(project)
+        break
+      default:
+        throw new Error(`unexpected husky rule ${ruleId}`)
+    }
+  })
+}
+
+const sourceFolder = path.join(__dirname, '../../..', 'src/common/git/assets')
+const templateFile = 'template'
+const templateString = '{{COMMAND}}'
+const destinationFolder = '.husky'
+
+const checkCargo = (project: NodeProject, options: CheckCargoOptions) => {
```

Comment:
@quesabe I would keep the rules separately in `husky/<rule name>/`.

Reply:
I left the rules within this module intentionally - this way a user has no way to call this function on his own. Separating into several files we leave a possibility to import the function and call it.
Overall it should not be a big problem unless someone wants to hack the system, so let's divide it into several files.

Reply:
@quesabe yeah, agree, doesn't look like a big problem to me too. OI would suggest moving to separate files in favour of cleaner structure and better decomposition.